### ISSUE TITLE
CI/CD pipeline for project

### ISSUE THREAD

**Issue by dominic0df** [amosproj/amos2024ss08-cloud-native-llm#23]
## User story
1. As a developer
2. I want to implement a CI/CD pipeline for our project
3. So that tests are automatically executed, the bill of materials is updated

## Acceptance criteria
* tests are automatically run
* the project is build
* the bill of materials is updated
* optional - flag if dependency has no permissive license

## Further information
* It is possible to automatically run a build for each merge requests
* It is possible to enforce that a merge request must build before it can be merged

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
